packages/plugin-edgedb/src/edgedb-field-builder.ts

```diff
@@ -0,0 +1,189 @@
+const RootBuilder: {
+  // eslint-disable-next-line @typescript-eslint/prefer-function-type
+  new <Types extends SchemaTypes, Shape, Kind extends FieldKind>(
+    name: string,
+    builder: PothosSchemaTypes.SchemaBuilder<Types>,
+    kind: FieldKind,
+    graphqlKind: PothosSchemaTypes.PothosKindToGraphQLType[FieldKind],
+  ): PothosSchemaTypes.RootFieldBuilder<Types, Shape, Kind>;
+} = RootFieldBuilder as never;
+
+export class EdgeDBObjectFieldBuilder<
+  Types extends SchemaTypes,
+  Model extends
+    | ({ [ModelKey in keyof Model]: Model[ModelKey] extends infer U ? U : never } & {
+        Fields: string | never;
+        Links: {
+          [Key in Model['Fields']]: { Shape: Model['Links'][Key] };
+        };
+      })
+    | never,
+  Shape extends object = Exclude<Model, 'Links' & 'Fields'>,
+> extends RootBuilder<Types, Model, 'EdgeDBObject'> {
+  model: string;
+  edgeDBFieldMap: FieldMap;
+
+  exposeBoolean = this.createExpose('Boolean');
+  exposeFloat = this.createExpose('Float');
+  exposeInt = this.createExpose('Int');
+  exposeID = this.createExpose('ID');
+  exposeString = this.createExpose('String');
+  exposeBooleanList = this.createExpose(['Boolean']);
+  exposeFloatList = this.createExpose(['Float']);
+  exposeIntList = this.createExpose(['Int']);
+  exposeIDList = this.createExpose(['ID']);
+  exposeStringList = this.createExpose(['String']);
+
+  constructor(
+    name: string,
+    builder: PothosSchemaTypes.SchemaBuilder<Types>,
+    model: string,
+    fieldMap: FieldMap,
+  ) {
+    super(name, builder, 'EdgeDBObject', 'Object');
+
+    this.model = model;
+    this.edgeDBFieldMap = fieldMap;
+  }
+
+  link<
+    Field extends Model['Fields'],
+    Nullable extends boolean,
+    Args extends InputFieldMap,
+    ResolveReturnShape,
+  >(
+    ...allArgs: NormalizeArgs<
+      [
+        name: Field,
+        options?: RelatedFieldOptions<
+          Types,
+          Model,
+          Field,
+          Nullable,
+          Args,
+          ResolveReturnShape,
+          Shape
+        >,
+      ]
+    >
+  ): FieldRef<Model['Links'][Field]['Shape'], 'Object'> {
+    const [name, { ...options } = {} as never] = allArgs;
+    const relationField = getLink(this.model, this.builder, name);
+    const ref = getRefFromModel(extractTargetTypeName(relationField.target), this.builder);
+
+    const { resolve, extensions, ...rest } = options;
+
+    const isList = isMultiLink(relationField);
+
+    console.log('[debug] link() -> name: ', name);
+    console.log('[debug] link() -> relationField: ', relationField, relationField.target);
+    console.log('[debug] link() -> ref: ', ref);
+    console.log('[debug] link() -> isList: ', isList);
+
+    return this.field({
+      ...(rest as {}),
+      type: isList ? [ref] : ref,
+      // TODO: Descriptions
+      // description: getFieldDescription(this.model, this.builder, name, description),
+      extensions: {
+        ...extensions,
+        pothosEdgeDBFallback:
```

Comment:
the fallback resolver stuff in pothos is one of my least favorite parts.  having a resolver option that only sometimes gets called is really confusing for users, so if this can be avoided for edgeDB that would be cool.  I haven't seen if you have started working on any entity-reloading logic yet.

Reply:
Haven't really looked into it, just copy left from the prisma plugin. Will take a look into it, we could avoid it. No entity-reloading logic existing yet.